BREAKING_CHANGES.md

```diff
@@ -32,12 +32,21 @@
 *   The `BranchPrefixToTrim` configuration property has been removed. `RegularExpression` is now used to capture named groups instead.
     *   Default `RegularExpression` for feature branches is changed from `^features?[/-]` to `^features?[/-](?<BranchName>.+)` to support using `{BranchName}` out-of-the-box
     *   Default `RegularExpression` for unknown branches is changed from `.*` to `(?<BranchName>.*)` to support using `{BranchName}` out-of-the-box
+*   The `Mainline` mode and the related implementation has been removed completely. The new `TrunkBased` version strategy should be used instead.
 *   The branch related property `is-mainline` in the configuration system has been renamed to `is-main-branch`
 *   The versioning mode has been renamed to deployment mode and consists of following values:
     *   ManualDeployment (previously ContinuousDelivery)
     *   ContinuousDelivery (previously ContinuousDeployment)
     *   ContinuousDeployment (new)
-	
+*   On the configuration root level a new property with name `version-strategy` has been introduced with following values:
+    *   ConfigNext
+    *   MergeMessage
+    *   TaggedCommit
+    *   TrackReleaseBranches
```

Comment:
I'm a bit unsure about the name `TrackReleaseBranches`. We should be consistent with whether we use a verb (`Track`) in the enum values  and whether we use singular (`Commit`) or plural (`Branches`) form.

Reply:
I'm open to any renaming as long we are renming the underlying class who implements the strety as well.

Reply:
Is `TrackReleaseBranches` really a versioning strategy, though? Isn't it more like a branch configuration?

Reply:
At the moment it is a version strategy yes. What is your suggestion!? To delete it? I thought this is the concept of our core domain to have provider based (strategy based) approach to determine the next version. And you can control whether or not it is called using the strategies property on the configuration root level.

Reply:
Yes, I like the configurable strategy approach. But like `next-version`, `tracks-release-branches` is a configuration property. It's strange to have `tracks-release-branches` configured for a branch, but ignored because `TrackReleaseBranches` isn't configured.

Reply:
Also here I would say like I already did previously: You need to think of the property like a characteristic of an branch or a characteristic of the repository which is generally isolated of the actual used strategies. If someone decide e.g. not to use the NextVersionVersionStrategy but define next-version then it would be fine to ignore it.

And you second example is even clear to argument. The track-release-brachnes property will be used in the TrackReleaseBranchesVersionStrategy and TrunkBaseVersionStrategy. If you decide to use TrunkBaseVersionStrategy you should not execute the TrackReleaseBranchesVersionStrategy because it gives you maybe a wrong behavior for the TrunkBased (Mainline) workflow.

Does it makes sense?

Reply:
I didn't think of the possibility of `tracks-release-branches` being used by more than one strategy. That's not the case for `next-version`, is it?

Reply:
Sorry I was mistaken. I have mixed `tracks-release-branches` with `track-merge-message` in my mind. You are right `tracks-release-branches` will be used in `TrackReleaseBranchesVersionStrategy` and `next-version` in `ConfiguredNextVersionVersionStrategy` only.